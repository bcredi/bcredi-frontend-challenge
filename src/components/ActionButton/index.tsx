import React from "react";
import { ReactComponent as LockSvg } from "../../assets/lock-solid.svg";
import styled from "styled-components";

interface props {
  text: string;
  type: "submit" | "reset";
}

const ActionButton = ({ text, type }: props) => {
  return (
    <Button type={type}>
      <LockIcon />
      <p>{text}</p>
    </Button>
  );
};

const Button = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 320px;
  height: 48px;
  background-color: var(--base-color-link);
  border: none;
  border-radius: 3px;
  padding: 14px 10px;
  color: var(--base-color-white);
  cursor: pointer;
  margin-top: 48px;
  margin-bottom: auto;
  position: relative;
  transition: all ease-out 0.1s;

  &:hover {
    background-color: var(--base-color-link-hover);
    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.3);
  }

  &: focus {
    outline: 0;
  }

  > svg {
    position: absolute !important;
  }

  > p {
    font-weight: bold;
    font-size: 16px;
    line-height: 20px;
  }

  @media (max-width: 600px) {
    width: 312px;
  }
`;

const LockIcon = styled(LockSvg)`
  width: 10px;
  height: 12.3px;
  position: relative;
  left: 31px;
`;

export default ActionButton;
