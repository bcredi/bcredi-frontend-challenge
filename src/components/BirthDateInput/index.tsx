import React, { useState } from "react";
import styled from "styled-components";
import { InputError, Input } from "../../styles";

interface props {
  setEmpty: React.Dispatch<React.SetStateAction<boolean>>;
  error: boolean;
  handleChange: any;
  value: string;
}

function validateDateFormat(date: string) {
  if (date !== "") {
    const match = date.match(
      /^([1-9]|0[1-9]|[12][0-9]|3[01])[/]([1-9]|0[1-9]|1[012])[/](19|20)\d\d$/
    );
    const correctFormat = match !== null;

    if (correctFormat) {
      const currentYear = new Date().getFullYear();
      const monthMaxDays = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
      const [day, month, year] = date.split("/").map((s) => +s); // Conversão para int

      if (year <= currentYear) {
        // months - fev
        if (month === 1 || month > 2) {
          if (day > monthMaxDays[month - 1]) {
            return "day outside the month limits";
          } else {
            return 1;
          }
        }

        // fev
        if (month === 2) {
          const leapYear =
            (!(year % 4) && year % 100) || (!(year % 400) as boolean);
          if (!leapYear && day >= 29) {
            return "day outside the limits of February - not a leap year";
          } else if (leapYear && day > 29) {
            return "day outside the limits of February - leap year";
          } else {
            return 1;
          }
        }
      } else {
        return "year bigger than the current one";
      }
    } else {
      return "incorrect format";
    }
  } else {
    return "empty date";
  }
}

const BirthDateInput = ({ setEmpty, error, handleChange, value }: props) => {
  const [validBirthDate, setValidBirthDate] = useState(true);

  function validateDate(date: string) {
    const dateValidationStatus = validateDateFormat(date);
    if (dateValidationStatus === 1) {
      setEmpty(false);
      return setValidBirthDate(true);
    } else if (dateValidationStatus === "empty date") {
      setValidBirthDate(true);
    } else {
      setEmpty(false);
      return setValidBirthDate(false);
    }
  }

  return (
    <InputWrapper>
      <InputLabel>Data de nascimento</InputLabel>
      <StyledInput
        required
        type="text"
        placeholder="DD/MM/AAAA"
        maxLength={10}
        name="birthDate"
        value={value}
        onChange={(e) => {
          e.target.value = e.target.value.replace(
            /^(\d{2})(\d{2})(\d{4})$/,
            "$1/$2/$3"
          );
          handleChange(e);
        }}
        onBlur={(e) => validateDate(e.target.value)}
        onInvalid={() => setEmpty(true)}
        style={{
          border: `1px solid ${
            error || !validBirthDate
              ? "var(--base-color-error)"
              : "var(--base-color-border)"
          }`,
          color: `${
            error || !validBirthDate
              ? "var(--base-color-error"
              : "var(--base-color-text)"
          }`,
        }}
      />
      <StyledInputError>
        {!validBirthDate
          ? "Data inválida"
          : error
          ? "O campo é obrigatório"
          : null}
      </StyledInputError>
    </InputWrapper>
  );
};

const InputWrapper = styled.div`
  width: 148px;
  display: flex;
  flex-direction: column;
  position: relative;

  @media (max-width: 600px) {
    width: 312px;
  }
`;

const InputLabel = styled.label`
  width: 148px;
  font-size: 14px;
  line-height: 18px;
  color: var(--base-color-warm-grey);
  text-align: left;

  @media (max-width: 600px) {
    width: 312px;
    margin-top: 30px;
  }
`;

const StyledInput = styled(Input)`
  width: auto;
`;

const StyledInputError = styled(InputError)`
  top: 78px;

  @media (max-width: 600px) {
    top: 114px;
  }
`;

export { BirthDateInput, validateDateFormat };
