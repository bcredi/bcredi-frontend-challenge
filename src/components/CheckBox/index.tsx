import React from "react";
import styled from "styled-components";
import { ReactComponent as CheckSvg } from "../../assets/check-solid.svg";
import { InputError } from "../../styles";

interface props {
  setNotChecked: React.Dispatch<React.SetStateAction<boolean>>;
  error: boolean;
  handleChange: React.Dispatch<React.SetStateAction<boolean>>;
  value: boolean;
}

const Checkbox = ({ setNotChecked, handleChange, error, value }: props) => {
  function toggleCheck() {
    if (!value) {
      handleChange(true);
      setNotChecked(false);
    } else {
      handleChange(false);
      setNotChecked(true);
    }
  }

  return (
    <InputWrapper
      style={{
        border: `1px solid ${
          error ? "var(--base-color-error)" : "var(--base-color-border)"
        }`,
      }}
    >
      <FakeCheckBox
        style={{ backgroundColor: value ? "var(--base-color-link)" : "" }}
        onClick={toggleCheck}
      >
        {value && <CheckIcon />}
      </FakeCheckBox>
      <InputInfo>
        Li e estou de acordo com a{" "}
        <a href="/privacy-policy" title="Política de privacidade - Bcredi">
          Política de Privacidade
        </a>{" "}
        e a{" "}
        <a href="/use-policy" title="Política de uso - Bcredi">
          Política de Uso de Informações
        </a>
        .
      </InputInfo>
      {error && (
        <StyledInputError>Por favor confirme os termos de uso</StyledInputError>
      )}
    </InputWrapper>
  );
};

const FakeCheckBox = styled.div`
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid var(--base-color-link);
  border-radius: 4px;
  cursor: pointer;
`;

const StyledInputError = styled(InputError)`
  top: 60px;
  @media (max-width: 600px) {
    top: 97px;
  }
`;

const InputInfo = styled.div`
  width: 267px;
  margin: 7px 5px 9px 12px;
  font-size: 12px;
  line-height: 20px;
  color: var(--base-color-pinkish-grey);

  > a {
    color: var(--base-color-link);
    text-decoration: none;
  }

  @media (max-width: 600px) {
    width: 230px;
  }
`;

const InputWrapper = styled.div`
  height: 56px;
  width: 320px;
  border-radius: 3px;
  border: 1px solid var(--base-color-border);
  padding: 7px 5px 9px 12px;
  text-align: left;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  position: relative;
  margin-top: 36px;

  > input {
    width: 24px;
    height: 24px;
    border-radius: 4px;
    cursor: pointer;
  }

  @media (max-width: 600px) {
    width: 312px;
    height: 92px;
  }
`;

const CheckIcon = styled(CheckSvg)`
  width: 9.7px;
  height: 7.1px;
  color: var(--base-color-white);
`;

export default Checkbox;
