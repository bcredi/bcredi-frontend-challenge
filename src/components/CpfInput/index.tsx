import React, { useState } from "react";
import styled from "styled-components";
import { InputError, Input } from "../../styles";

interface props {
  setEmpty: React.Dispatch<React.SetStateAction<boolean>>;
  error: boolean;
  handleChange: any;
  value: string;
}

const rejectedCpfs = [
  "00000000000",
  "11111111111",
  "22222222222",
  "33333333333",
  "44444444444",
  "55555555555",
  "66666666666",
  "77777777777",
  "88888888888",
  "99999999999",
];

function validateCpfFormat(cpf: string) {
  if (cpf !== "") {
    const match = cpf.match(/\d\d\d[.]\d\d\d[.]\d\d\d[-]\d\d$/);
    const correctFormat = match !== null;
    if (correctFormat) {
      const stringOnlyNumbers = cpf.replace(/[.-]/g, "");
      if (rejectedCpfs.includes(stringOnlyNumbers)) {
        return "repetitive numbers";
      } else {
        const cpfNumbersArray = stringOnlyNumbers.split("").map((n) => +n);

        const sumNumberFirstDigit = cpfNumbersArray
          .slice(0, 9)
          .map((n, index) => n * (10 - index))
          .reduce((a, b) => a + b, 0);

        const sumNumberSecondDigit = cpfNumbersArray
          .slice(0, 10)
          .map((n, index) => n * (11 - index))
          .reduce((a, b) => a + b, 0);

        var restDigitOne = (sumNumberFirstDigit * 10) % 11;
        var restDigitTwo = (sumNumberSecondDigit * 10) % 11;

        if (restDigitOne === 10 || restDigitOne === 11) {
          restDigitOne = 0;
        }
        if (restDigitTwo === 10 || restDigitTwo === 11) {
          restDigitTwo = 0;
        }
        if (
          restDigitOne !== cpfNumbersArray[9] ||
          restDigitTwo !== cpfNumbersArray[10]
        ) {
          return "rest does not match the digits";
        } else {
          return 1;
        }
      }
    } else {
      return "incorrect format";
    }
  } else {
    return "empty cpf";
  }
}

const CpfInput = ({ setEmpty, error, handleChange, value }: props) => {
  const [validCpf, setValidCpf] = useState(true);

  function validateCpf(cpf: string) {
    const cpfValidationStatus = validateCpfFormat(cpf);
    if (cpfValidationStatus === 1) {
      setEmpty(false);
      return setValidCpf(true);
    } else if (cpfValidationStatus === "empty cpf") {
      setValidCpf(true);
    } else {
      setEmpty(false);
      return setValidCpf(false);
    }
  }

  return (
    <InputWrapper>
      <InputLabel>CPF</InputLabel>
      <StyledInput
        required
        placeholder="000.000.000-00"
        maxLength={14}
        name="cpf"
        value={value && value}
        onChange={(e) => {
          e.target.value = e.target.value.replace(
            /^(\d{3})(\d{3})(\d{3})(\d{2})$/,
            "$1.$2.$3-$4"
          );
          handleChange(e);
        }}
        onBlur={(e) => validateCpf(e.target.value)}
        onInvalid={() => setEmpty(true)}
        style={{
          border: `1px solid ${
            error || !validCpf
              ? "var(--base-color-error)"
              : "var(--base-color-border)"
          }`,
          color: `${
            error || !validCpf
              ? "var(--base-color-error"
              : "var(--base-color-text)"
          }`,
        }}
      />
      <StyledInputError>
        {!validCpf
          ? "CPF não é válido"
          : error
          ? "O campo é obrigatório"
          : null}
      </StyledInputError>
    </InputWrapper>
  );
};

const InputWrapper = styled.div`
  width: 148px;
  display: flex;
  flex-direction: column;
  position: relative;

  @media (max-width: 600px) {
    width: 312px;
  }
}
`;

const InputLabel = styled.label`
  width: 148px;
  font-size: 14px;
  line-height: 18px;
  color: var(--base-color-warm-grey);
  text-align: left;

  @media (max-width: 600px) {
    width: 312px;
  }
`;

const StyledInput = styled(Input)`
  width: auto;
`;

const StyledInputError = styled(InputError)`
  top: 78px;

  @media (max-width: 600px) {
    top: 84px;
  }
`;

export { CpfInput, validateCpfFormat, rejectedCpfs };
