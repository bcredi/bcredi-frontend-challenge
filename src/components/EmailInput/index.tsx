import React, { useState } from "react";
import styled from "styled-components";
import { InputWrapper, InputError, Input } from "../../styles";

interface props {
  setEmpty: React.Dispatch<React.SetStateAction<boolean>>;
  error: boolean;
  handleChange: any;
  value: string;
}

/* function validate(email: string) {
  if (email !== "") {
    const user = email.substring(0, email.indexOf("@"));
    const domain = email.substring(email.indexOf("@") + 1, email.length);
    if (
      user.length >= 1 &&
      user.search("@") === -1 &&
      user.search(" ") === -1 &&
      domain.length >= 3 &&
      domain.search("@") === -1 &&
      domain.search(" ") === -1 &&
      domain.search(".") !== -1 &&
      domain.indexOf(".") >= 1 &&
      domain.lastIndexOf(".") < domain.length - 1
    ) {
      return 1;
    } else {
      return "incorrect format";
    }
  } else {
    return "empty email";
  }
} */

function validateEmailFormat(email: string) {
  if (email !== "") {
    const match = email.match(
      // eslint-disable-next-line no-useless-escape
      /^(([^<>()[\]\\.,+`^*!'¨|;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
    );
    const correctFormat = match !== null;
    if (correctFormat) {
      return 1;
    } else {
      return "incorrect format";
    }
  } else {
    return "empty email";
  }
}

const EmailInput = ({ setEmpty, error, handleChange, value }: props) => {
  const [validEmail, setValidEmail] = useState(true);

  function validateEmail(email: string) {
    const emailValidationStatus = validateEmailFormat(email);
    if (emailValidationStatus === 1) {
      setEmpty(false);
      return setValidEmail(true);
    } else if (emailValidationStatus === "empty email") {
      setValidEmail(true);
    } else {
      setEmpty(false);
      return setValidEmail(false);
    }
  }

  return (
    <StyledInputWrapper>
      <InputLabel>E-mail</InputLabel>
      <Input
        required
        type="text"
        className="email-input"
        placeholder="ana.maria@email.com"
        name="email"
        value={value}
        onChange={handleChange}
        maxLength={60}
        onBlur={(e) => validateEmail(e.target.value)} // Quando o input perde o foco
        onInvalid={() => setEmpty(true)}
        style={{
          border: `1px solid ${
            error || !validEmail
              ? "var(--base-color-error)"
              : "var(--base-color-border)"
          }`,
          color: `${
            error || !validEmail
              ? "var(--base-color-error"
              : "var(--base-color-text)"
          }`,
        }}
      />
      <InputError>
        {!validEmail
          ? "O e-mail informado não é válido"
          : error
          ? "O campo e-mail é obrigatório"
          : null}
      </InputError>
    </StyledInputWrapper>
  );
};

const InputLabel = styled.label`
  font-size: 14px;
  line-height: 18px;
  color: var(--base-color-warm-grey);
  text-align: left;
`;

const StyledInputWrapper = styled(InputWrapper)`
  margin-top: 0;

  @media (max-width: 600px) {
    margin-top: 0;
  }
`;

export { EmailInput, validateEmailFormat };
