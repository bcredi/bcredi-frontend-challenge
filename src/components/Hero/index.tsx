import React from "react";
import styled from "styled-components";

interface props {
  image: string;
  quote: string;
  author: string;
  company: string;
}

const Hero = ({ image, quote, author, company }: props) => {
  const ImageWrapper = styled.div`
    height: 100vh;
    width: 43.203125%;
    background-image: linear-gradient(
        rgba(0, 0, 0, 0.2),
        rgba(0, 0, 0, 0) 16%,
        rgba(0, 0, 0, 0) 42%,
        rgba(0, 0, 0, 0.5) 69%,
        rgba(0, 0, 0, 0.8) 96%
      ),
      url(${image});
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    position: relative;

    @media (max-width: 600px) {
      display: none;
    }
  `;

  return (
    <ImageWrapper>
      <HeroContent>
        <HeroQuote>{quote}</HeroQuote>
        <HeroAuthorInfo>{`${author},`}</HeroAuthorInfo>
        <HeroAuthorInfo>{company}</HeroAuthorInfo>
      </HeroContent>
    </ImageWrapper>
  );
};

const HeroContent = styled.div`
  color: var(--base-color-white);
  text-align: right;
  width: 66.85%;
  position: absolute;
  right: 7.25%;
  bottom: 4.72%;
`;
const HeroQuote = styled.p`
  margin-bottom: 8px;
  font-size: 20px;
  line-height: 30px;
`;
const HeroAuthorInfo = styled.p`
  font-size: 16px;
  line-height: 24px;
`;

export default Hero;
