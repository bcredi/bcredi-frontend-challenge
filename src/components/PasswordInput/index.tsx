import React, { useState } from "react";
import styled from "styled-components";
import { InputWrapper, InputError, Input } from "../../styles";
import { ReactComponent as EyeSvg } from "../../assets/eye-solid.svg";
import { ReactComponent as EyeSlashSvg } from "../../assets/eye-slash-solid.svg";

interface props {
  setEmpty: React.Dispatch<React.SetStateAction<boolean>>;
  error: boolean;
  handleChange: any;
  value: string;
}

function validatePasswordFormat (password: string) {
  if( password !== "") {
    if(password.length >= 8){
      return 1;
    } else {
      return "incorrect format";
    }
  } else {
    return "empty password";
  }
}

const PasswordInput = ({ setEmpty, error, handleChange, value }: props) => {
  const [validPassword, setValidPassword] = useState(true);
  const [visible, toggleVisible] = useState(false);

  function validatePassword(value: string) {
    if (value !== "") {
      setEmpty(false);
      if (value.length < 8) {
        setValidPassword(false);
      } else {
        setValidPassword(true);
      }
    } else {
      setValidPassword(true);
    }
  }

  return (
    <InputWrapper>
      <InputLabel>Senha</InputLabel>
      <Input
        required
        placeholder="Cadastre uma senha"
        type={visible ? "text" : "password"}
        minLength={8}
        name="password"
        value={value && value}
        onChange={(e) => {
          handleChange(e);
        }}
        onBlur={(e) => validatePassword(e.target.value)}
        onInvalid={() => setEmpty(true)}
        style={{
          border: `1px solid ${
            error || !validPassword
              ? "var(--base-color-error)"
              : "var(--base-color-border)"
          }`,
          color: `${
            error || !validPassword
              ? "var(--base-color-error"
              : "var(--base-color-text)"
          }`,
        }}
      />
      {!visible ? (
        <EyeIcon onClick={() => toggleVisible(!visible)} />
      ) : (
        <EyeSlashIcon onClick={() => toggleVisible(!visible)} />
      )}
      <InputError>
        {!validPassword
          ? "A senha precisa ter no mínimo 8 caracteres"
          : error
          ? "O campo senha é obrigatório"
          : null}
      </InputError>
    </InputWrapper>
  );
};

/* const StyledInputWrapper = styled(InputWrapper)`
  input[type="password"]::-ms-reveal,
  input[type="password"]::-ms-clear {
    display: none;
  }
`; */

const InputLabel = styled.label`
  width: 148px;
  font-size: 14px;
  line-height: 18px;
  color: var(--base-color-warm-grey);
  text-align: left;

  @media (max-width: 600px) {
    width: 312px;
  }
`;

const EyeIcon = styled(EyeSvg)`
  width: 16px;
  height: 10px;
  position: relative;
  color: var(--base-color-warm-grey);
  margin-left: -30px;
  cursor: pointer;
  &:hover {
    color: var(--base-color-text);
  }
`;

const EyeSlashIcon = styled(EyeSlashSvg)`
  width: 16px;
  height: 10px;
  position: relative;
  color: var(--base-color-warm-grey);
  margin-left: -30px;
  cursor: pointer;
  &:hover {
    color: var(--base-color-text);
  }
`;

export { PasswordInput, validatePasswordFormat } ;
