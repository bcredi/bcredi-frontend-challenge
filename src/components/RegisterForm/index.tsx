import React, { FormEvent, ReactNode } from "react";
import { ReactComponent as LockSvg } from "../../assets/lock-solid.svg";
import styled from "styled-components";
import { User, InputError } from "../../pages/Register/types";

interface props {
  formTitle: string;
  formDescription: string;
  handleSubmit: () => void;
  children?: ReactNode;
  footer?: ReactNode;
  setData: React.Dispatch<React.SetStateAction<User>>;
  data: User;
  setFormErrors: React.Dispatch<React.SetStateAction<InputError>>;
  formErrors: InputError;
  readCheck: boolean;
}

const RegisterFormContainer = ({
  formTitle,
  formDescription,
  handleSubmit,
  children,
  footer,
  readCheck,
  formErrors,
  setFormErrors,
  setData,
  data,
}: props) => {
  function resetForm() {
    setData({} as User);
    setFormErrors({} as InputError);
  }

  console.log("USER DATA: ", data); // PRINT
  console.log("Aceito os termos: ", readCheck); // PRINT

  const onSubmit = (e: FormEvent) => {
    e.preventDefault();
    if (!data.email || data.email === "") {
      setFormErrors({ ...formErrors, emailError: true });
    } else if (!data.cpf || data.cpf === "") {
      setFormErrors({ ...formErrors, cpfError: true });
    } else if (!data.birthDate || data.birthDate === "") {
      setFormErrors({ ...formErrors, birthDateError: true });
    } else if (!data.password || data.password === "") {
      setFormErrors({ ...formErrors, passwordError: true });
    } else if (!readCheck) {
      setFormErrors({ ...formErrors, readCheckError: true });
    } else {
      handleSubmit();
      resetForm();
    }
  };

  return (
    <RegisterFormWrapper>
      <FormTitle>{formTitle}</FormTitle>
      <FormText>{formDescription}</FormText>
      <Form name="register" onSubmit={onSubmit}>
        {children}
        <ActionButton type="submit">
          <LockIcon />
          <p>Cadastrar</p>
        </ActionButton>
      </Form>
      <FormFooter>{footer}</FormFooter>
    </RegisterFormWrapper>
  );
};

const FormTitle = styled.div`
  width: 312px;
  height: 30px;
  font-weight: bold;
  font-size: 24px;
  line-height: 30px;
  color: var(--base-color-text);

  @media (min-width: 600px) {
    width: 320px;
    text-align: center;
  }
`;

const FormText = styled.div`
  height: 54px;
  font-size: 16px;
  line-height: 24px;
  color: var(--base-color-warm-grey);
  margin: 6px auto 0px auto;

  @media (min-width: 600px) {
    text-align: center;
  }
`;

const FormFooter = styled.div`
  width: 320px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid var(--base-color-border);
  margin-top: 20px;
  margin-bottom: 16px;
  padding-top: 16px;

  > p {
    font-size: 14px;
    line-height: 18px;
    color: var(--base-color-warm-grey);
  }

  p > a {
    color: var(--base-color-link);
    text-decoration: none;
  }

  @media (max-width: 600px) {
    width: 312px;
    margin-top: 81px;
    padding-top: 15.8px;
  }
`;

const RegisterFormWrapper = styled.div`
  width: 320px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 38px;
  margin-bottom: 40px;

  @media (max-width: 600px) {
    width: 312px;
    margin-top: 84px;
    margin-bottom: 32px;
  }
`;

const Form = styled.form`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  padding-top: 23px;
`;

const ActionButton = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 320px;
  height: 48px;
  background-color: var(--base-color-link);
  border: none;
  border-radius: 3px;
  padding: 14px 10px;
  color: var(--base-color-white);
  cursor: pointer;
  margin-top: 48px;
  margin-bottom: auto;
  position: relative;
  transition: all ease-out 0.1s;

  &:hover {
    background-color: var(--base-color-link-hover);
    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.3);
  }

  &: focus {
    outline: 0;
  }

  > svg {
    position: absolute !important;
  }

  > p {
    font-weight: bold;
    font-size: 16px;
    line-height: 20px;
  }

  @media (max-width: 600px) {
    width: 312px;
  }
`;

const LockIcon = styled(LockSvg)`
  width: 10px;
  height: 12.3px;
  position: relative;
  left: 31px;
`;

export default RegisterFormContainer;
