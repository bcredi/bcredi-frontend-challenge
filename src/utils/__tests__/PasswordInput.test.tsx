import React from "react";
import { PasswordInput } from "../../components/PasswordInput";
import renderer from "react-test-renderer";

test("Checks if the component is correctly rendered", () => {
  const component = renderer
    .create(<PasswordInput setEmpty={() => {}} error={false} handleChange={() => {}} value={""} />)
    .toJSON();

  expect(component).toMatchSnapshot();
});
