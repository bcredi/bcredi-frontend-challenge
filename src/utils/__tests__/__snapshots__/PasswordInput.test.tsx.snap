// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks if the component is correctly rendered 1`] = `
<div
  className="sc-eCssSg jhDcFB"
>
  <label
    className="sc-jrAGrp iUlFgE"
  >
    Senha
  </label>
  <input
    className="sc-jSgupP jHzLmv"
    minLength={8}
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    onInvalid={[Function]}
    placeholder="Cadastre uma senha"
    required={true}
    style={
      Object {
        "border": "1px solid var(--base-color-border)",
        "color": "var(--base-color-text)",
      }
    }
    type="password"
    value=""
  />
  <svg
    className="sc-kEjbxe gKFWaf"
    onClick={[Function]}
  >
    eye-solid.svg
  </svg>
  <div
    className="sc-gKsewC dGAaCy"
  />
</div>
`;
